libsolidity/formal/SMTLib2Interface.cpp

```diff
@@ -214,6 +218,13 @@ vector<string> SMTLib2Interface::parseValues(string::const_iterator _start, stri
 
 string SMTLib2Interface::querySolver(string const& _input)
 {
+	if (m_smtSolverCallback)
```

Comment:
Shouldn't the logic be to first check if it is part of the query responses, then ask the callback (if present) and add it to the query responses otherwise? Also - do we want to add callback failures to the unhandled queries?